### DISCUSSION TITLE
Share Your Serverless Articles!

### DISCUSSION THREAD

**Discussion by nitya** [Azure/Cloud-Native#2] [Serverless September: Community Buzz]
## About Community Buzz

Have you written any interesting articles or tutorials showcasing Azure technologies like Azure Functions, Azure Container Apps, Dapr etc.? We would love to feature and amplify selected content in September **and beyond** so the broader community can learn from your experiences!

 * Use [custom issue](https://github.com/Azure/Cloud-Native/issues/new?assignees=&labels=&template=community-buzz--technical-articles.md&title=) - to submit short questions for the sessions, in advance
 * We will create "Community Buzz: Roundup" posts on our blog on a periodic basis (bi-weekly or monthly) and feature selected submissions from the list. Your issue will be updated to reflect the roundup post where it was featured.

**Comment by ambarishg** [comment]
Please have a look at https://github.com/ambarishg/paddy-doctor.          

This is a good example where we combine a number of #azure services
#azurecustomvision      
#azurecontainerapp         
#azurekeyvault           
#azuremanagedidentity

**Comment by tjoudeh** [comment]
Hi @nitya, I've created a detailed tutorial of 10 posts where I build step-by-step a tasks management application following the microservices architecture pattern. This application will consist of 3 microservices, each with specific capabilities to show how Azure Container Apps and Dapr can simplify the building of a microservice application.

Various Dapr building blocks are used in this tutorial such as:

- Service to Service invocation:  Frontend Web App microservice invokes the Backend Web API microservice using Dapr sidecar.
- State Management: Backend API stores Azure Cosmos DB data and stores email logs on Azure Table Storage using Dapr State Management building blocks.
- Pub/Sub: Backend API publishes messages to Azure Service Bus when a task is saved and a Backend processor microservices consumes those messages and sends emails using SendGrid.
- Bindings: The backend processor is triggered based on an incoming event such as a Cron job. Integration with external resources such as Azure Storage Queue and Invoking external resources using output bindings.

IaC code using Bicep and GitHub actions are in place for easier recreation of the complete resources needed for students/learners

**The blog posts can be accessed from [this link](https://bit.ly/ACATutorial), and the code repository is available [here](https://github.com/tjoudeh/TasksTracker.ContainerApps)**

**List of posts within the tutorial:**
1.	[Tutorial for building Microservice Applications with Azure Container Apps and Dapr – Part 1](https://bitoftech.net/2022/08/25/tutorial-building-microservice-applications-azure-container-apps-dapr/)
2.	[Deploy backend API Microservice to Azure Container Apps – Part 2](https://bitoftech.net/2022/08/25/deploy-microservice-application-azure-container-apps/)
3.	[Communication between Microservices in Azure Container Apps – Part 3](https://bitoftech.net/2022/08/25/communication-microservices-azure-container-apps/)
4.	[Dapr Integration with Azure Container Apps – Part 4](https://bitoftech.net/2022/08/29/dapr-integration-with-azure-container-apps/)
5.	[Azure Container Apps State Store With Dapr State Management API – Part 5](https://bitoftech.net/2022/08/29/azure-container-apps-state-store-with-dapr-state-management-api/)
6.	[Azure Container Apps Async Communication with Dapr Pub/Sub API – Part 6](https://bitoftech.net/2022/09/02/azure-container-apps-async-communication-with-dapr-pub-sub-api-part-6/)
7.	[Azure Container Apps with Dapr Bindings Building Block – Part 7](https://bitoftech.net/2022/09/05/azure-container-apps-with-dapr-bindings-building-block/)
8.	Monitor Azure Container Apps with Application Insights – Part 8 Coming Soon
9.	Integrate Health probes in Azure Container Apps – Part 9 Coming Soon
10.	Azure Container Apps Auto Scaling with KEDA – Part 10 Coming Soon

**Architecture diagram of the microservices application:**
![image](https://user-images.githubusercontent.com/3114431/188848087-913f691e-9b65-4303-972a-280509afe82e.png)
